File: src/main/java/com/stockmanager/services/CategoriaService.java
package com.stockmanager.services;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.stockmanager.entities.Categoria;
import com.stockmanager.repositories.CategoriaRepository;

@Service
public class CategoriaService {

	@Autowired
	private CategoriaRepository categoriaRepository;
	
	public List<Categoria> findAll(){
		return (List<Categoria>) this.categoriaRepository.findAll();
	}
	
	public Categoria save(Categoria categoria){
		Categoria saved = this.categoriaRepository.save(categoria);
		return saved;
	}
	
	public void delete(Long id){
		this.categoriaRepository.delete(id);
	}
}

File: src/main/webapp/resources/core/js/services/categoria.service.js
app.factory('CategoriaService', ['$http', function($http){
	
	var _findAll = function(){
		return $http.get(SERVER_APP + '/categoria/findall');
	}
	
	var _save = function(categoria){
		return $http.post(SERVER_APP + '/categoria/', categoria);
	}
	
	var _delete = function(id){
		return $http.get(SERVER_APP + '/categoria/delete/' + id);
	}
	
	return {
		
		findAll: _findAll,
		
		save: _save,
		
		delete: _delete
		
	}
}])